lib/files.js

```diff
@@ -35,22 +38,42 @@ function createImportHook() {
  */
 function createSyncImportHook() {
   let imports = {};
-  return function (fpath, kind, cb) {
-    fpath = path.resolve(fpath);
+  return function ({path: fpath, importerPath}, cb) {
+    fpath = path.resolve(path.dirname(importerPath), fpath);
     if (imports[fpath]) {
       cb();
     } else {
       imports[fpath] = true;
-      cb(null, fs.readFileSync(fpath, {encoding: 'utf8'}));
+      cb(null, {
+        contents: fs.readFileSync(fpath, {encoding: 'utf8'}),
+        path: fpath
+      });
     }
   };
 }
 
+/**
+ * Check if the given input string is "path-like" or a path to an existing file,
+ * and if so, read it. This requires it to contain a path separator
+ * (`path.sep`). If not, this will return `null`.
+ */
+function tryReadFileSync(str) {
+  if (
+    typeof str == 'string' &&
+    str.indexOf(path.sep) !== -1 &&
+    fs.existsSync(str)
+  ) {
+    // Try interpreting `str` as path to a file.
+    return fs.readFileSync(str, {encoding: 'utf8'});
```

Comment:
Optional: the `existsSync` call could be omitted by catching `readFileSync` errors and checking the error's code.

Reply:
I've done so